apicast/src/oauth/keycloak.lua

```diff
@@ -237,9 +237,7 @@ function _M:authorize(service, client)
   end
 
   local url = resty_url.join(self.config.authorize_url, ngx.var.is_args, ngx.var.args)
-  local res = http_client.get(url)
-
-  _M.respond_and_exit(res.status, res.body, res.headers)
+  ngx.redirect(url)
```

Comment:
Yes. No proxying of the IDP from the Gateway. 

Gateway will connect to the IDP to download it's public key so it can verify JWT later and that is the only time it actually connects to the IDP.

Caller is supposed to obtain the Bearer token from the IDP by other means. Gateway is not involved in the process.